src/validation.cpp

```diff
@@ -1338,18 +1376,33 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
             results.emplace(wtxid, MempoolAcceptResult::MempoolTxDifferentWitness(iter.value()->GetTx().GetWitnessHash()));
         } else {
             // Transaction does not already exist in the mempool.
-            txns_new.push_back(tx);
+            // Try submitting the transaction on its own.
+            const auto single_res = AcceptSingleTransaction(tx, single_args);
```

Comment:
I think the `TxValidationState` could be inspected here. If the failure cannot be assigned to the feerate check, the package validation could fail early ? Otherwise if you have a package of 25th component and the latest is faultive, we might validate twice the first 24 ones. Once here, the other one in `AcceptMultipleTransactions` and this could be avoid.

Am I correct ?

If yes, I think it's always good to minimize our DoS surface. Though I'm unsure if the difference would be significant between the two implementations.

Reply:
Correct - if the failure isn't feerate or missing inputs (i.e. the child, or the parents could depend on each other), then there's nothing package validation can do. I had this in the original version and took it out, but you're right that it's safer to have here. I've added it back in the latest push.